### PR TITLE
grip-grab: init at 0.3.0

### LINKED ISSUE #346651: Package request: `grip-grab`
**Project description**

A fast, more lightweight ripgrep alternative for daily use cases.

**Metadata**

* homepage URL: https://github.com/alexpasmantier/grip-grab
* source URL: https://github.com/alexpasmantier/grip-grab
* license: unlicense
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
Closes #346651
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by griffi-gh** [comment]
(perhaps the self-upgrade functionality should be disabled or patched out? i opened an issue upstream too: https://github.com/alexpasmantier/grip-grab/issues/6)

**Comment by griffi-gh** [comment]
Also, the latest release is missing a git tag, crates.io has 0.5.4 at commit c79886354cdadfb7939c7c659ac894b44a654b03

File: pkgs/by-name/gr/grip-grab/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  cargoHash = "sha256-NEoZYdZEhT4gPMm9fUn3WbFPIcjVhLLFxEvbQxVHV+k=";
+
+  meta = {
+    description = "Fast, more lightweight ripgrep alternative for daily use cases";
+    homepage = "https://github.com/alexpasmantier/grip-grab";
+    changelog = "https://github.com/alexpasmantier/grip-grab/releases";
+    license = lib.licenses.unlicense;
+    maintainers = with lib.maintainers; [ luftmensch-luftmensch ];
+    mainProgram = "grip-grap";
```
Comment (griffi-gh) [inline pkgs/by-name/gr/grip-grab/package.nix:28]: Shouldn't this be `gg`?  
Unless it also provides `grip-grab` as the executable name

Comment (luftmensch-luftmensch) [reply pkgs/by-name/gr/grip-grab/package.nix:28]: You're right. It was a typo. Fixing now

File: pkgs/by-name/gr/grip-grab/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    hash = "sha256-5nKDeJ5d9ECd9zvSK22zN15m8qCt/L3VKem9hVQzLzU=";
+  };
+
+  cargoHash = "sha256-NEoZYdZEhT4gPMm9fUn3WbFPIcjVhLLFxEvbQxVHV+k=";
+
+  meta = {
+    description = "Fast, more lightweight ripgrep alternative for daily use cases";
+    homepage = "https://github.com/alexpasmantier/grip-grab";
+    changelog = "https://github.com/alexpasmantier/grip-grab/releases";
+    license = lib.licenses.unlicense;
```
Comment (Atemu) [inline pkgs/by-name/gr/grip-grab/package.nix:26]: Where did this come from? It's Apache-2.0 AFAICT.

**Review by Atemu — COMMENTED** [review]
The diff is quite minimal and LGTM apart from this.

How significant is the self-updater? These need to be disabled in most cases as they're likely to misbehave with Nix.

@griffi-gh could you comment on whether the package works as expected?

File: pkgs/by-name/gr/grip-grab/package.nix (hunk 2 above)
Comment (luftmensch-luftmensch) [reply pkgs/by-name/gr/grip-grab/package.nix]: I didn't double checked it. Fixed

**Review by Atemu — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 346658`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>grip-grab</li>
  </ul>
</details>

**Comment by Atemu** [comment]
Ah, I would like the thing about the auto-updater to be cleared up though. Could you explain?

File: pkgs/by-name/gr/grip-grab/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  version = "0.3.0";
+in
+rustPlatform.buildRustPackage {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "alexpasmantier";
+    repo = "grip-grab";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-5nKDeJ5d9ECd9zvSK22zN15m8qCt/L3VKem9hVQzLzU=";
```
Comment (FliegendeWurst) [inline pkgs/by-name/gr/grip-grab/package.nix:17]:
```suggestion
  version = "0.6.7";
in
rustPlatform.buildRustPackage {
  inherit pname version;

  src = fetchFromGitHub {
    owner = "alexpasmantier";
    repo = "grip-grab";
    tag = "v${version}";
    hash = "sha256-e7duLL4tjW+11jXUqU6sqoKTAPGkH81iDCfjtNcnd4I=";
```
This also resolves the auto-upgrade functionality (it is disabled by default).

File: pkgs/by-name/gr/grip-grab/package.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  cargoHash = "sha256-NEoZYdZEhT4gPMm9fUn3WbFPIcjVhLLFxEvbQxVHV+k=";
```
Comment (FliegendeWurst) [inline pkgs/by-name/gr/grip-grab/package.nix:20]:
```suggestion
  cargoHash = "sha256-g7YdNNN6YJiW8sQa1KMj5evVbLPObRXrKM5HoG87Kxc=";

  doCheck = false; # no tests
```
Test configuration takes quite some time to compile for no benefit.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by Atemu** [comment]
What about the internal updater?

**Comment by FliegendeWurst** [comment]
It is now a cargo feature, disabled by default. https://github.com/alexpasmantier/grip-grab/blob/main/Cargo.toml#L35-L37

**Review by Atemu — APPROVED** [review]
This LGTM then.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**